File: FortyTwo/Shared/Models/Bid.cs
namespace FortyTwo.Shared.Models
{
    public enum Bid
    {
        Pass = 0,
        Thirty = 30,
        ThirtyOne = 31,
        ThirtyTwo = 32,
        ThrityThree = 33,
        ThirtyFour = 34,
        ThrityFive = 35,
        ThritySix = 36,
        ThritySeven = 37,
        ThrityEight = 38,
        ThirtyNine = 39,
        Fourty = 40,
        FourtyOne = 41,
        FourtyTwo = 42,
        EightyFour = 84,
        ThreeMarks = 126,
        FourMarks = 168,
        Plunge = 169,
        FiveMarks = 210,
        SixMarks = 252,
        SevenMarks = 294
    }

    public static class BidExtensions
    {
        public static string ToPrettyString(this Bid bid)
            => bid switch
            {
                Bid.Pass => bid.ToString(),
                Bid.Plunge => bid.ToString(),
                Bid.ThreeMarks => "3 Marks",
                Bid.FourMarks => "4 Marks",
                Bid.FiveMarks => "5 Marks",
                Bid.SixMarks => "6 Marks",
                Bid.SevenMarks => "7 Marks",
                _ => ((int)bid).ToString()
            };

        public static string ToPrettyString(this Bid? bid)
            => bid == null || !bid.HasValue
                ? "N/A"
                : bid.Value.ToPrettyString();
    }
}

File: cloudflare/packages/rules/src/bid.ts
export enum Bid {
  Pass = 0,
  Thirty = 30,
  ThirtyOne = 31,
  ThirtyTwo = 32,
  ThirtyThree = 33,
  ThirtyFour = 34,
  ThirtyFive = 35,
  ThirtySix = 36,
  ThirtySeven = 37,
  ThirtyEight = 38,
  ThirtyNine = 39,
  Forty = 40,
  FortyOne = 41,
  FortyTwo = 42,
  EightyFour = 84,
  ThreeMarks = 126,
  FourMarks = 168,
  Plunge = 169,
  FiveMarks = 210,
  SixMarks = 252,
  SevenMarks = 294
}

export function bidToPrettyString(bid: Bid | null | undefined): string {
  if (bid === null || bid === undefined) return 'N/A';
  return bid === Bid.Pass || bid === Bid.Plunge
    ? Bid[bid]
    : bid === Bid.ThreeMarks
      ? '3 Marks'
      : bid === Bid.FourMarks
        ? '4 Marks'
        : bid === Bid.FiveMarks
          ? '5 Marks'
          : bid === Bid.SixMarks
            ? '6 Marks'
            : bid === Bid.SevenMarks
              ? '7 Marks'
              : String(bid);
}
